src/fsharp/ErrorLogger.fs

```diff
@@ -294,9 +285,12 @@ module ErrorLoggerExtensions =
         | :? System.IO.IOException -> () // This covers FileNotFoundException and DirectoryNotFoundException
         | :? System.UnauthorizedAccessException -> ()
         | Failure _ // This gives reports for compiler INTERNAL ERRORs
+#if FX_NO_REDUCEDEXCEPTIONS
+#else
         | :? System.SystemException -> 
             PreserveStackTrace(exn)
             raise exn
+#endif
```

Comment:
Net result is that when `FX_NO_REDUCEDEXCEPTIONS` is defined, this function always returns unit. If that's the intention, easiest to just put the #if at the top and leave the rest of the code unchanged in the #else.